src/panels/panels.tsx

```diff
@@ -1,12 +1,29 @@
 import { useQueryParam, StringParam } from "use-query-params";
 import { AnimatePresence } from "framer-motion";
+import useEventListener from "hooks/useEventListener";
 
 import RegisterController from "components/RegisterController/RegisterController";
 
 import "./_panels.scss";
 
+// Close panel if Escape key is pressed when panel active
+export const close = {
+  onEscape: function (
+    e: KeyboardEvent,
+    queryStringSetter: (qs: undefined) => void
+  ) {
+    if (e.code === "Escape") {
+      queryStringSetter(undefined);
+    }
+  },
+};
+
 export default function Panels() {
-  const panelQs = useQueryParam("panel", StringParam)[0];
+  const [panelQs, setPanelQs] = useQueryParam("panel", StringParam);
+
+  useEventListener("keydown", (e: KeyboardEvent) =>
+    close.onEscape(e, setPanelQs)
```

Comment:
I'm not sure we should land this escape code as-is as we know it's going to cause issues right away if the panel content cannot prevent it from happening. 

I'd prefer it to not close on escape until we have an implementation that is cancellable.